pkg/storage/utils/decomposedfs/xattrs/xattrs.go

```diff
@@ -74,3 +82,30 @@ const (
 	UserAcePrefix  string = "u:"
 	GroupAcePrefix string = "g:"
 )
+
+// TODO this does not belong here
+var uuidRegexp = "[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}"
+
+// ReferenceFromAttr returns a CS3 reference from xattr of a node.
+// Supported formats are: "cs3:storageid/nodeid"
+func ReferenceFromAttr(b []byte) (*provider.Reference, error) {
```

Comment:
if you enforce i uuid, how do we import a storage space from oc10? create a uuid for every file? might be something we have to do anyway. I guess this is related to a logical nodeid mepping to an internal storage specific nodeid see https://github.com/owncloud/ocis/pull/2318/files#diff-77ff7f7b179b132bd53b54fb5d3d1f016baa4e8930e6d719cea6b894c9b51f6bR101-R114